src/plugins/administrative_templates/admx/booleanvalue.h

```diff
@@ -0,0 +1,100 @@
+/***********************************************************************************************************************
+**
+** Copyright (C) 2021 BaseALT Ltd. <org@basealt.ru>
+**
+** This program is free software; you can redistribute it and/or
+** modify it under the terms of the GNU General Public License
+** as published by the Free Software Foundation; either version 2
+** of the License, or (at your option) any later version.
+**
+** This program is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program; if not, write to the Free Software
+** Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+**
+***********************************************************************************************************************/
+
+#ifndef GPUI_BOOLEAN_VALUE_H
+#define GPUI_BOOLEAN_VALUE_H
+
+#include <string>
+#include <stdexcept>
+
+namespace model {
+namespace admx {
+/*!
+ * \brief Describes a value that is equivalent to some logical state (true, false)
+ */
+class BooleanValue
+{
+public:
+    /*!
+     * \brief Enumeration of all possible value types
+     */
+    enum Type {
+        BOOLEAN_VALUE_TYPE_DELETED,
+        BOOLEAN_VALUE_TYPE_DECIMAL,
+        BOOLEAN_VALUE_TYPE_STRING,
+    };
+
+    /*!
+     * \brief Construct BooleanValue with deleted value.
+     */
+    BooleanValue();
+    /*!
+     * \brief Construct BooleanValue with decimal value.
+     */
+    BooleanValue(unsigned int decimalValue);
+    /*!
+     * \brief Construct BooleanValue with string value
+     */
+    BooleanValue(const std::string &string);
```

Comment:
While this  is generally it is good idea to have methods and constructors in structure. In this case we should remove them and focus on having only public members in a struct like in the rest of the structures in the administrative templates plugin.

Reply:
For now, this is a temporary solution. I am currently preparing a commit in which some of the fields of the structure will be in union, which will allow you to store structures more compactly in memory. Because of this, you will need to manage your lifetime. In this case, if you make the fields open, there will be a risk of reading "invalid memory", which may lead to incorrect operation of the program.